src/View/Helper/TidyHelper.php

```diff
@@ -138,25 +137,26 @@ public function tidyErrors($in = '', &$out = '')
         if (function_exists('tidy_parse_string')) {
             $tidy = tidy_parse_string($out, [], 'UTF8');
             $tidy->cleanRepair();
+            /** @psalm-suppress UndefinedPropertyFetch */
             $errors = $tidy->errorBuffer . "\n";
 
             return $errors;
         }
 
         // cli
-        $File = new File(rtrim(TMP, DS) . DS . rand() . '.html', true);
-        $File->write($out);
-        $path = $File->pwd();
-        $errors = $path . '.err';
-        $this->_exec("tidy -eq -utf8 -f $errors $path");
-        $File->delete();
-
-        if (!file_exists($errors)) {
+        $filePath = rtrim(TMP, DS) . DS . uniqid() . '.html';
+        file_put_contents($filePath, $out);
+        $errorFilePath = $filePath . '.err';
+        $this->_exec("tidy -eq -utf8 -f $errorFilePath $filePath");
```

Comment:
`TMP` could contain spaces, so arguments should be escaped.

But anyway, I only find reference to this class in `src/View/AjaxView.php` via `@property`, but no further usage. This class can be deleted.

Reply:
Added f6a4479 2012-04-14 Tidy Panel
Removed 5c20afa 2012-05-12 Remove tidy panel

Reply:
Yeah it did seem to me that the class was unused but I kept my focus to just fixing errors reported by psalm.

I can remove it if @markstory confirms.

Reply:
I think this was intended as an optional panel. I personally don't use it, perhaps @dereuromark can speak to how useful he finds it?

Reply:
I must say I didnt use it much at all :)

Reply:
We could remove it and see if anyone notices :smile:

Reply:
Done